client/src/app/routes/stardust/landing/Landing.tsx

```diff
@@ -189,6 +202,15 @@ class Landing extends Feeds<RouteComponentProps<LandingRouteProps>, LandingState
             const milestones = this._feedClient.getItems()
                 .filter(item => item.payloadType === "MS").slice(0, MAX_MILESTONE_ITEMS);
 
+            if (this.state.latestMilestones.length > 0) {
+                milestones.push(...this.state.latestMilestones);
+                if (milestones.length > MAX_MILESTONE_ITEMS) {
+                    this.setState({
+                        latestMilestones: this.state.latestMilestones.slice(0, this.state.latestMilestones.length - 1)
```

Comment:
This looks confusion, why do we need a state for `latestMilestones` and for `milestones` separately ?